### PR TITLE
#107

### LINKED ISSUE #107: null values and addConcreteSettersForOptional
I find that addConcreteSettersForOptional uses `Optional#of` instead of `Optional#ofNullable` quite surprising especially since Intellij IDEA by default doesn't warn against passing nulls to `javax.validation.constraints.NotNull` annotated parameters.
What is the reasoning behind this decision?

### PR DESCRIPTION
**Pull request by lpandzic**
Fixes #107

### CODE REVIEW

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java [hunk 1]
```diff
@@ -993,12 +993,21 @@ public MyRecordBuilder p(T p) {
...
-                .addStatement("this.$L = $T.of($L)", component.name(), type.typeName(), component.name());
+                .addStatement(getOptionalStatement(type), component.name(), type.typeName(), component.name());
         addConstructorAnnotations(component, parameterSpecBuilder);
         methodSpec.addStatement("return this").addParameter(parameterSpecBuilder.build());
         builder.addMethod(methodSpec.build());
     }
 
+    private String getOptionalStatement(OptionalType type) {
+
+        if(type.typeName().equals(TypeName.get(Optional.class))) {
```
[the remark this answers is not in the document]
Comment (lpandzic) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java:1004]: 1. isOptional is private
2. how would I get `ClassType` from `OptionalType`?

[the remark this answers is not in the document]
Comment (Randgalt) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java:1004]: Add this to `OptionalType`:

```
    public boolean isOptional()
    {
        return typeName.equals(optionalType);
    }
```

[the remark this answers is not in the document]
Comment (lpandzic) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java]: done

**Comment by Randgalt** [comment]
Thanks for the PR

**Pull request merged**